Hospital/HospitalApi/Controllers/RoomController.cs

```diff
@@ -44,5 +44,17 @@ public Room UpdateRoom(Room room)
             var roomRepo = _uow.GetRepository<IRoomWriteRepository>();
             return roomRepo.Update(room);
         }
+
+
+        [HttpGet("find")]      
+        public IActionResult FindByName([FromQuery(Name = "roomName")] string roomName, [FromQuery(Name = "buildingName")] string buildingName)
+        {
+            var roomRepo = _uow.GetRepository<IRoomReadRepository>();
+            if (roomName==null)
+            {
+                return BadRequest();
+            }
```

Comment:
You are returning BadRequest only in case of roomName being null. What if buildingName is null?
Pay attention to code formatting (roomName == null instead of roomName==null)